### ISSUE TITLE
Future Settings: support for lists on configuration files

### ISSUE THREAD

**Issue by clebergnu** [avocado-framework/avocado#3935]
I can not see to get list values applied from the configuration file.  Example for tests (adding to `selftests/unit/test_future_settings.py`:

```python
example = """[foo]
bar = default from file
non_registered = this should be ignored
space_separated_list = one two three
evaled_list = ['one', 'two', 'three']
"""

...


    def test_list_space_separated(self):
        stgs = settings.Settings()
        stgs.process_config_path(self.config_file.name)
        stgs.register_option(section='foo',
                             key='space_separated_list',
                             key_type=list,
                             default=[],
                             help_msg='just a test')
        config = stgs.as_dict()
        result = config.get('foo.space_separated_list')
        self.assertIsInstance(result, list)
        self.assertEqual(3, len(result))

    def test_list_evaled(self):
        stgs = settings.Settings()
        stgs.process_config_path(self.config_file.name)
        stgs.register_option(section='foo',
                             key='evaled_list',
                             key_type=list,
                             default=[],
                             help_msg='just a test')
        config = stgs.as_dict()
        result = config.get('foo.evaled_list')
        self.assertIsInstance(result, list)
        self.assertEqual(3, len(result))
```

What's the right way to do it?

**Comment by beraldoleal** [comment]
Thanks @clebergnu I will take a look.

This is also related to #3435

**Comment by beraldoleal** [comment]
@clebergnu I did a test with `run.loaders` that is already implemented, and today, we already support this, using the following formats:

1. Using comma for separation (without spaces)

```ini
[run]
loaders = file,tap
```

2. Using lists syntax and separed by commas (also without spaces):

```ini
[run]
loaders = [file,tap]
```

Both cases were converted to python lists, I double-checked using `avocado config | grep run.loaders`.

I understand that we could improve here, but will this help you, for now?

**Comment by clebergnu** [comment]
@beraldoleal, yes those work. The tricky part is that the current `list` values on the configuration files are quoted, as a real Python list would be. Examples:

```
loaders = ['file', '@DEFAULT']

[plugins.resolver]
order = ['avocado-instrumented', 'python-unittest', 'glib', 'robot', 'exec-test']
```

Because they are eval'ed. So the issue about how to define them still remains, IMO.  Also IMO, we should remove the eval'ed implementation, specially the current version that is a fake eval (the list with names that are not strings).

**Comment by beraldoleal** [comment]
@clebergnu ok, I will keep that in mind when working on #3435

**Comment by beraldoleal** [comment]
@clebergnu I'm thinking about this. Just one question: When you say: " the current version that is a fake eval" are you talking about `core/settings.py:206`?

**Comment by clebergnu** [comment]
> @clebergnu I'm thinking about this. Just one question: When you say: " the current version that is a fake eval" are you talking about `core/settings.py:206`?

Well, that's a slightly "better" eval, for what's worth it.

What I meant about "fake" is that on `avocado.core.future.settings` it takes square brackets, like a non-fake list would, but it takes what would be a variable name in a real Python list, and not strings, because they're not quoted.  So,

 * `[foo, bar]`: Fake
 * `["foo", "bar"]`: Not fake

And yes, the "fake" wording was confusing, I apologize.

**Comment by willianrampazzo** [comment]
So, I reached the same problem while working on the migration of `enabled_extension_manager.py`. Functional tests on `test_basic.py`, `PluginsTest` started to fail because they build custom configuration files with content like:

```
config_content = "[plugins]\ndisable=['cli.cmd.sysinfo',]" 
```

And to make some of them work, I need to change to:

```
config_content = "[plugins]\ndisable=[cli.cmd.sysinfo,]" 
```

Before I change all the tests, I would like to know which path we will take. If we still accept true Python lists as arguments in the config, I'll hold the changes until a fix is available.

**Comment by beraldoleal** [comment]
@willianrampazzo , yes hold on, please. I'm working to bring back python lists support.

**Comment by willianrampazzo** [comment]
> @willianrampazzo , yes hold on, please. I'm working to bring back python lists support. PR is on the way.

You rock sir!

**Comment by beraldoleal** [comment]
This was fixed by #3939
